docs/src/sway-program-types/smart_contracts.md

```diff
@@ -81,29 +65,22 @@ Note that the above implementation of the ABI follows the [Checks, Effects, Inte
 
 ## Calling a Smart Contract from a Script
 
+>**Note**: In most cases, calling a contract should be done from the [Rust SDK](../testing/testing-with-rust.md) or the [TypeScript SDK](../frontend/typescript_sdk.md) which provide a more ergonomic UI for interacting with a contract. However, there are situations where manually writing a script to call a contract is required.
+
 Now that we have defined our interface and implemented it for our contract, we need to know how to actually _call_ our contract. Let's take a look at a contract call:
 
 ```sway
-script;
-
-use std::consts::BASE_ASSET_ID;
-
-use wallet_abi::Wallet;
-use wallet_abi::SendFundsRequest;
-
-fn main() {
-    let contract_address = 0x9299da6c73e6dc03eeabcce242bb347de3f5f56cd1c70926d76526d7ed199b8b;
-    let caller = abi(Wallet, contract_address);
-    let amount_to_send = 200;
-    let recipient_address: 0x9299da6c73e6dc03eeabcce242bb347de3f5f56cd1c70926d76526d7ed199b8b;
-    caller.send_funds{gas: 10000, coins: 0, asset_id: BASE_ASSET_ID}(amount_to_send, recipient_address);
-}
+{{#include ../../../examples/wallet_contract_caller_script/src/main.sw}}
 ```
 
 The main new concept is the _abi cast_: `abi(AbiName, contract_address)`. This returns a `ContractCaller` type which can be used to call contracts. The methods of the ABI become the methods available on this contract caller: `send_funds` and `receive_funds`. We then directly call the contract ABI method as if it was just a regular method. You also have the option of specifying the following special parameters inside curly braces right before the main list of parameters:
 
-1. `gas`: represents the gas being forwarded to the contract when it is called.
-2. `coins`: represents how many coins are being forwarded with this call.
-3. `asset_id`: represents the ID of the _asset type_ of the coins being forwarded.
+1. `gas`: a `u64` that represents the gas being forwarded to the contract when it is called.
+2. `coins`: a `u64` that represents how many coins are being forwarded with this call.
+3. `asset_id`: a `b256` that represents the ID of the _asset type_ of the coins being forwarded.
```

Comment:
I always feel that 'which' is better than 'that' in cases like this, but I'm not sure if one or the other is more correct or whether it matters, really.
```suggestion
1. `gas`: a `u64` which represents the gas being forwarded to the contract when it is called.
2. `coins`: a `u64` which represents how many coins are being forwarded with this call.
3. `asset_id`: a `b256` which represents the ID of the _asset type_ of the coins being forwarded.
```